src/main/java/com/newrelic/telemetry/opentelemetry/export/NewRelicSpanExporter.java

```diff
@@ -36,12 +35,12 @@
    * @param spanBatchSender An instance that sends a SpanBatch to the New Relic trace ingest API
    * @since 0.1.0
    */
-  NewRelicSpanExporter(SpanBatchAdapter adapter, SpanBatchSender spanBatchSender) {
-    if (spanBatchSender == null) {
-      throw new IllegalArgumentException("You must provide a non-null SpanBatchSender");
+  NewRelicSpanExporter(SpanBatchAdapter adapter, TelemetryClient telemetryClient) {
```

Comment:
Can you address the _why_ behind this api change?  It seems like a lower-level thing now depends on a higher-level thing.  That is, it always seemed to me like the TelemetryClient was the highest API for users of this exporter.  There's now some mixing of those strata/boundaries.

I'm curious what the intention is with this.

Reply:
Exporter calls TelemetryClient calls Senders

Exporter is responsible for converting from "native" representation to ours
Telemetry client is unified API for spans/metrics and handles retries.
Senders serialize data and send to NR APIs.

I think this is the right abstraction organization.

Previously, the NRSE was creating it's own TelemetryClient...now it can be reused between Spans and Metrics.

Reply:
👍